documentation/docs/developer_guide/coding_standards.md

```diff
@@ -1 +1,179 @@
+
+We proceed assuming that the "issue" raised requires adding a new and significant feature to CABLE, requiring the addition of new files/modules to CABLE. Lesser modifications are covered under this assumption anyway.
+
+For discussion purposes we assume that you are including a new model to do some new stuff. This can conveniently be added in a single file. 
+
+## Code Template 
+
+Assuming the *new* model is called from eg_driver.F90.
+
+```fortran
+PROGRAM eg_driver()
+
+USE new_mod_cbl, ONLY : new
+
+IMPLICIT NONE
+Header Declarations
+...
+CALL eg_subr( Arguments )
+...
+
+END PROGRAM eg_driver
+```
+Where,
+
+```fortran
+MODULE eg_mod_cbl
+! Module description in markdown - e.g. [cable_roughness.F90](https://github.com/CABLE-LSM/CABLE/blob/main/src/science/roughness/cable_roughness.F90) for an example, which is then redered [here](https://cable.readthedocs.io/en/latest/api/module/cable_roughness_module.html)
+
+PUBLIC :: eg_subr 
+PRIVATE
+
+CONTAINS
+
+SUBROUTINE eg_subr ( Arguments )
+...
+END SUBROUTINE eg_subr 
+
+END MODULE eg_mod_cbl
+```
+
+Where,
+
+```fortran
+SUBROUTINE eg_subr( eg_arg, ilen, jlen, sometype )
+! Subroutine description in markdown - see??
+
+USE some_type_mod_cbl, ONLY : some_type
+
+IMPLICIT NONE
+
+INTEGER, INTENT(IN) :: ilen              ! example arg length
+INTEGER, INTENT(IN) :: jlen              ! example arg length
+REAL, INTENT(OUT) :: eg_arg(ilen,jlen)   ! example arg [kg//m^2/s]
+
+!local variables
+REAL :: eg_local(ilen,jlen)              ! example local [kg//m^2/s]
+
+DO i=1,ilen                         ! DO:over ilen
+
+  DO j=1,jlen                       ! DO:over jlen
+  
+    IF( condition1 ) THEN           ! IF:condition1  
+  
+      IF( condition1.1 ) THEN       ! IF:condition1.1  
+        
+        eg_arg = some_type%some_type_member1 * some_type%some_type_member2     &
+                * some_type%some_type_member3 
+      
+      ENDIF                         ! IF:condition1.1  
+      
+    ELSE
+        
+      IF( condition1.2 ) THEN       ! IF:condition1.2  
+      
+        eg_arg = some_type%some_type_member1 * some_type%some_type_member2     &
+                * some_type%some_type_member4 
+      
+      ENDIF                         ! IF:condition1.2    
+                          
+    ENDIF                           ! IF:condition1
+  
+  END DO                            ! DO:over jlen
+
+END DO                              ! DO:over ilen
```

Comment:
Why not use construct names here (for all the loops and if statements) instead? This way the names are actually useful and the compiler can warn us if something isn't right.

```suggestion
ilen_loop: DO i=1,ilen

  jlen_loop: DO j=1,jlen
  
    condition1: IF( condition1 ) THEN
  
      condition1.1: IF( condition1.1 ) THEN  
        
        eg_arg = some_type%some_type_member1 * some_type%some_type_member2     &
                * some_type%some_type_member3 
      
      END IF condition1.1
      
    ELSE
        
      condition1.2: IF( condition1.2 ) THEN 
      
        eg_arg = some_type%some_type_member1 * some_type%some_type_member2     &
                * some_type%some_type_member4 
      
      END IF condition1.2
                          
    END IF condition1
  
  END DO jlen_loop

END DO ilen_loop
```